backend/src/data/repositories/chat-message/chat-message.repository.ts

```diff
@@ -0,0 +1,109 @@
+import { SortOrder } from '~/common/enums/enums';
+import {
+  ChatMessageCreateRequestDto,
+  ChatMessageGetAllItemResponseDto,
+  ChatMessageGetRequestDto,
+} from '~/common/types/types';
+import {
+  ChatMessage as ChatMessageM,
+  MenteesToMentors as MenteesToMentorsM,
+} from '~/data/models/models';
+
+type Constructor = {
+  ChatMessageModel: typeof ChatMessageM;
+  MenteesToMentors: typeof MenteesToMentorsM;
+};
+
+class ChatMessage {
+  #ChatMessageModel: typeof ChatMessageM;
+
+  #MenteesToMentors: typeof MenteesToMentorsM;
+
+  public constructor({ ChatMessageModel, MenteesToMentors }: Constructor) {
+    this.#ChatMessageModel = ChatMessageModel;
+    this.#MenteesToMentors = MenteesToMentors;
+  }
+
+  public getAll({
+    userId,
+    chatOpponentId,
+  }: ChatMessageGetRequestDto): Promise<ChatMessageGetAllItemResponseDto[]> {
+    return this.#ChatMessageModel
+      .query()
+      .select()
+      .where({ senderId: userId, receiverId: chatOpponentId })
+      .orWhere({ senderId: chatOpponentId, receiverId: userId })
+      .withGraphJoined(
+        '[sender(withoutPassword).[userDetails], receiver(withoutPassword).[userDetails]]',
+      )
+      .castTo<ChatMessageGetAllItemResponseDto[]>()
+      .execute();
+  }
+
+  public create({
+    senderId,
+    receiverId,
+    message,
+  }: ChatMessageCreateRequestDto): Promise<ChatMessageGetAllItemResponseDto> {
+    return this.#ChatMessageModel
+      .query()
+      .insert({ senderId, receiverId, message })
+      .withGraphFetched(
+        '[sender(withoutPassword).[userDetails], receiver(withoutPassword).[userDetails]]',
+      )
+      .castTo<ChatMessageGetAllItemResponseDto>()
+      .execute();
+  }
+
+  public async getAllLastMessages(
+    userId: number,
+  ): Promise<ChatMessageGetAllItemResponseDto[]> {
+    const usersMentors = await this.#MenteesToMentors
+      .query()
+      .select()
+      .where({ menteeId: userId })
+      .returning('mentees_to_mentors.mentor_id')
```

Comment:
Why don't you just select needed cols inside select?
```suggestion
      .select('mentees_to_mentors.mentor_id')
      .where({ menteeId: userId })
```